### PR TITLE
refactor(builder): change output of `version` subcommand.

### PR DESCRIPTION
**Pull request by devoc09**
With this change, running `builder version` on binaries installed with `go install` will output the version specified at the suffix.

If it's under development, the following output will be obtained.
```
$ GO111MODULE=on CGO_ENABLED=0 go run . version
ocb version (devel)
```

**Link to tracking Issue:** #8770

### CODE REVIEW

File: cmd/builder/internal/version.go [hunk 1]
```diff
@@ -5,22 +5,37 @@ package internal // import "go.opentelemetry.io/collector/cmd/builder/internal"
...
+}
+
 func versionCommand() *cobra.Command {
 	return &cobra.Command{
 		Use:   "version",
 		Short: "Version of ocb",
 		Long:  "Prints the version of the ocb binary",
 		Run: func(cmd *cobra.Command, args []string) {
-			cmd.Println(fmt.Sprintf("%s version %s", cmd.Parent().Name(), version))
+			cmd.Println(fmt.Sprintf("%s version %s", cmd.Parent().Name(), binVersion()))
```
Comment (bogdandrutu) [inline cmd/builder/internal/version.go:38]: Use `RunE` and return an error?

File: cmd/builder/internal/version.go [hunk 2]
```diff
@@ -5,22 +5,37 @@ package internal // import "go.opentelemetry.io/collector/cmd/builder/internal"
...
 
+// binVersion returns the version of the binary.
+// If the version is not set, it attempts to read the build information.
+func binVersion() string {
+	if version != "" {
+		return version
+	}
+	info, ok := debug.ReadBuildInfo()
+	if !ok {
+		log.Fatal("Failed to read build info")
```
Comment (bogdandrutu) [inline cmd/builder/internal/version.go:27]: Return an error instead of crashing.

File: cmd/builder/internal/version.go (hunk 1 above)
Comment (devoc09) [reply cmd/builder/internal/version.go]: Thank you for the review.
I have changed it to use `RunE` in order to return an error.

https://github.com/open-telemetry/opentelemetry-collector/pull/8994/files#diff-c3eff1de420450a694b19343e4543ca062f42962c968262c00d776504810861cR37-R41

File: cmd/builder/internal/version.go (hunk 2 above)
Comment (devoc09) [reply cmd/builder/internal/version.go]: Changed to return errro.

https://github.com/open-telemetry/opentelemetry-collector/pull/8994/files#diff-c3eff1de420450a694b19343e4543ca062f42962c968262c00d776504810861cR21-R29

**Comment by devoc09** [comment]
I have combined the commits into one and pushed it, so please check again.

**Review by bogdandrutu — APPROVED** [review]

**Comment by bogdandrutu** [comment]
Please add a changelog entry.

**Comment by devoc09** [comment]
> Please add a changelog entry.

Added changelog. Please review again 🙏

File: cmd/builder/internal/version.go [hunk 3]
```diff
@@ -5,22 +5,42 @@ package internal // import "go.opentelemetry.io/collector/cmd/builder/internal"
...
 import (
 	"fmt"
+	"runtime/debug"
 
 	"github.com/spf13/cobra"
 )
 
 var (
-	version = "dev"
+	version = ""
```
Comment (TylerHelmuth) [inline cmd/builder/internal/version.go:14]: Do we still need this variable?  I don't believe it is ever set, only read.  If thats true then the

```
if version != "" {
    return version, nil
}
```

is unneeded.

Comment (TylerHelmuth) [reply cmd/builder/internal/version.go:14]: I think we can replace any use of this variable with `binVersion()`

**Pull request merged**